features/lorawan/lorastack/mac/LoRaMac.cpp

```diff
@@ -1090,9 +1091,25 @@ lorawan_status_t LoRaMac::schedule_tx()
         _params.rx_window2_delay = _params.sys_params.join_accept_delay2
                                    + _params.rx_window2_config.window_offset;
     } else {
-        if (validate_payload_length(_params.tx_buffer_len,
+        // a check was performed for validity of FRMPayload in ::prepare_ongoing_tx() API.
+        // However, owing to the asynch nature of the send() API, we should check the
+        // validity again, as datarate may have changed since we last attempted to transmit.
+        if (_ongoing_tx_msg.type == MCPS_PROPRIETARY) {
+            frm_payload_size = _params.tx_buffer_len;
+        } else {
+            frm_payload_size = _params.tx_buffer_len - LORA_MAC_FRMPAYLOAD_OVERHEAD;
```

Comment:
overhead is a fudge factor for the get_max_payload having it pre-subtracted, so always needs to be applied, even for proprietary.